packages/router/addon/index.js

```diff
@@ -28,6 +28,20 @@ function lazyBundle(routeName, engineInfoByRoute) {
   return false;
 }
 
+function isEmberEngineOverridingRouter() {
+  // This router is intended to work with both classical and embroider
+  // builds. Currently, this collides with ember-engines/-private/router-ext's
+  // overrides. To overcome this, during embroider builds we remove that file
+  // from being included as apart of the build (and thus will not reopen the class).
+  // However, during classical builds this is not possible so here we check if that
+  // module is included and if it is we know that we are in a classical build and
+  // that the router is being reopened. If it is being reopened then we can simply
+  // call into ember-engines router as we dont need to do anything.
+
+  // eslint-disable-next-line no-undef
+  return !!requirejs.entries['ember-engines/-private/router-ext'];
```

Comment:
@ef4 This is the interesting piece of the whole PR. I have a couple of questions / thoughts about this for you.

1) Do you think this is the best approach for determining if ember-engines has reopened the router? I could also check the instance for things like `this._enginePromies` (which ember-engine adds) but that seems too brittle.

2) Should we simply noop all of our changes in the case that you are in classical mode?